src/pytypes/onefuzztypes/models.py

```diff
@@ -456,11 +456,22 @@ class NodeTasks(BaseModel):
     state: NodeTaskState = Field(default=NodeTaskState.init)
 
 
+class AutoScaleConfig(BaseModel):
+    image: str
+    max_size: Optional[int]  # max size of pool
```

Comment:
The level of data validation of this struct done in `pools/__init__.py` are reasonable, but should be done via pydantic validation `@validator` and `@root_validator`.

See JobConfig and NodeStateUpdate in this file for examples.

While the SDK doesn't make use of the validation available in the pydantic models yet, it would be great to make the same validation used at the service level happen at the SDK with minimal changes.

Reply:
Note, you won't be able to validate the image or region here.  The SDK doesn't have the ability to query for the underlying data.  Those still need to be done at the functions level.

Reply:
updated it to use validators